storage_service/locations/models/package.py

```diff
@@ -1097,39 +1130,44 @@ def finish_reingest(self, origin_location, origin_path, reingest_location, reing
+            # Update pointer file
+            root = etree.parse(self.full_pointer_file_path)
 
-        # Add compression event (if compressed)
-        amdsec = root.find('mets:amdSec', namespaces=utils.NSMAP)
-        if compression in (self.COMPRESSION_7Z_BZIP, self.COMPRESSION_7Z_LZMA):
-            try:
-                version = [x for x in subprocess.check_output('7z').splitlines() if 'Version' in x][0]
-                event_detail = 'program="7z"; version="{}"'.format(version)
-            except (subprocess.CalledProcessError, Exception):
-                event_detail = 'program="7z"'
-        elif compression in (self.COMPRESSION_TAR_BZIP2, self.COMPRESSION_TAR):
-            try:
-                version = subprocess.check_output(['tar', '--version']).splitlines()[0]
-                event_detail = 'program="tar"; version="{}"'.format(version)
-            except (subprocess.CalledProcessError, Exception):
-                event_detail = 'program="tar"'
-        else:
-            LOGGER.warning('Unknown compression algorithm, cannot correctly update pointer file')
-            event_detail = 'Unknown compression'
-        utils.mets_add_event(
-            amdsec,
-            'compression',
-            event_detail=event_detail,
-            event_outcome_detail_note=compress_output,
-        )
+            # Add compression event (if compressed)
+            amdsec = root.find('mets:amdSec', namespaces=utils.NSMAP)
+            if compression in (self.COMPRESSION_7Z_BZIP, self.COMPRESSION_7Z_LZMA):
+                try:
+                    version = [x for x in subprocess.check_output('7z').splitlines() if 'Version' in x][0]
+                    event_detail = 'program="7z"; version="{}"'.format(version)
+                except (subprocess.CalledProcessError, Exception):
+                    event_detail = 'program="7z"'
+            elif compression in (self.COMPRESSION_TAR_BZIP2, self.COMPRESSION_TAR):
+                try:
+                    version = subprocess.check_output(['tar', '--version']).splitlines()[0]
+                    event_detail = 'program="tar"; version="{}"'.format(version)
+                except (subprocess.CalledProcessError, Exception):
+                    event_detail = 'program="tar"'
+            else:
+                LOGGER.warning('Unknown compression algorithm, cannot correctly update pointer file')
+                event_detail = 'Unknown compression'
+            utils.mets_add_event(
+                amdsec,
+                'compression',
+                event_detail=event_detail,
+                event_outcome_detail_note=compress_output,
+            )
 
-        self.update_pointer_file(compression, root=root, path=out_path)
+            self.update_pointer_file(compression, root=root, path=out_path)
+        elif was_compressed:
+            # AIP used to be compressed, but is no longer so delete pointer file
+            os.remove(self.full_pointer_file_path)
```

Comment:
`full_pointer_file_path` is a read only property function. This should update `self.pointer_file_location` and `self.pointer_file_path`